test/tests/rackhd20/test_rackhd20_api_ucs.py

```diff
@@ -0,0 +1,297 @@
+                                   "password": "ucspe",
+                                   "ucs": self.UCS_IP,
+                                   "uri": "http://" + self.RACKHD_IP + ":7080/sys"
+                                   }
+                              }
+                        }
+        header = {"Content-Type": "application/json"}
+        api_data = fit_common.rackhdapi("/api/2.0/workflows", action="post",
+                                        headers=header, payload=data_payload)
+        id = api_data["json"]["context"]["graphId"]
+        self.assertEqual(api_data['status'], 201,
+                         'Incorrect HTTP return code, expected 201, got:' + str(api_data['status']))
+        self.wait_utility(str(id), 0, "Discovery")
+        api_data = fit_common.rackhdapi('/api/2.0/nodes')
+        self.assertEqual(api_data['status'], 200,
+                         'Incorrect HTTP return code, expected 200, got:' + str(api_data['status']))
+        logs.info_1("Found {0} Nodes after cataloging the UCS".format(len(api_data['json'])))
+
+        for node in api_data['json']:
+            if node["obms"] != [] and node["obms"][0]["service"] == "ucs-obm-service":
+                self.UCS_NODES.append(node)
+                if node["type"] == "compute":
+                    self.UCS_COMPUTE_NODES.append(node)
+        self.assertGreaterEqual(len(self.UCS_NODES), self.EXPECTED_UCS_NODES,
+                         'Expected to discover {0} UCS nodes, got: {1}'.format(self.EXPECTED_UCS_NODES, len(self.UCS_NODES)))
+        logs.info_1("Found {0} UCS nodes {1}".format(len(self.UCS_COMPUTE_NODES), self.UCS_COMPUTE_NODES))
+
+    @depends(after=test_api_20_ucs_discovery)
+    def test_api_20_ucs_catalog(self):
+        """
+        Tests the UCS Catalog workflow in rackHD
+        :return:
+        """
+        catalog_workflows = []
+        for x in xrange(len(self.UCS_NODES)):
+            postUrl = '/api/1.1/nodes/' + str(self.UCS_NODES[x]["id"]) + "/workflows?name=Graph.Ucs.Catalog"
+            header = {"Content-Type": "application/json"}
+            api_data = fit_common.rackhdapi(postUrl, headers=header, action="post")
+            self.assertEqual(api_data['status'], 201,
+                              'Expected to catalog {0} UCS nodes with status {1}, got: {2}'
+                              .format(self.UCS_NODES[x]["id"], 201, api_data['status']))
+            catalog_workflows.append(api_data["json"]["instanceId"])
+            logs.info_1("Posted URL: {0} with status: {1}".format(postUrl, api_data['status']))
+
+        # Restore the nodes, obms, and catalogs to their state before the UCS discovery
+        # in order to avoid any failure in other tests
+        logs.info_1("Restoring the database to the state it was in before the UCS discovery and catalog")
+        self.restore_node_utility(catalog_workflows)
+        self.restore_obms_utility()
+
+
+if __name__ == '__main__':
+    fit_common.unittest.main()
```

Comment:
after updating imports, change from fit_common.unittest.main() to unittest.main()